node/src/blockchain/blockchain_bridge.rs

```diff
@@ -314,25 +315,31 @@ impl BlockchainBridge {
             .get_block_number()
         {
             Ok(eb) => {
-                if u64::MAX == max_block_count {
+                if u64::MAX == max_block_count || u64::MAX == start_block_nbr {
                     BlockNumber::Number(eb)
                 } else {
                     BlockNumber::Number(eb.as_u64().min(start_block_nbr + max_block_count).into())
                 }
             }
             Err(e) => {
-                info!(
-                    self.logger,
-                    "Using 'latest' block number instead of a literal number. {:?}", e
-                );
```

Comment:
My opinion of the logging is that an error from the blockchain service provider doesn't have to be considered an error for the user if the application has a suitable resolution for it. Warning seemed too severe but Info didn't seem like too much. I understand there's a natural mismatch between logging the error at Info level, however, I think it would be worse to just ignore the error in case we receive a response we didn't expect. I can change it to DEBUG and remove the assertions for it.